tests/integration/polygonParts/polygonParts.spec.ts

```diff
@@ -5545,491 +5670,559 @@ describe('polygonParts', () => {
         });
 
-        it('should return 400 status code if polygonPartsEntityName is an invalid value - must follow a regex pattern (not more than 63 chars)', async () => {
-          const response = await requestSender.findPolygonParts({
-            params: { polygonPartsEntityName: 'a'.repeat(64) as EntityIdentifier },
-            body: { filter: featureCollection<Polygon | MultiPolygon>([]) },
-          });
+        expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+        expect(response).toSatisfyApiSpec();
 
-          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
-          expect(response).toSatisfyApiSpec();
+        expect.assertions(2);
+      });
 
-          expect.assertions(2);
+      it('should return 400 status code if polygonPartsEntityName is an invalid value - must end with raster product type', async () => {
+        const response = await requestSender.findPolygonParts({
+          params: { polygonPartsEntityName: 'invalid_name' as EntityIdentifier },
+          body: { filter: featureCollection<Polygon | MultiPolygon>([]) },
         });
 
-        it('should return 400 status code if polygonPartsEntityName is an invalid value - must end with raster product type', async () => {
-          const response = await requestSender.findPolygonParts({
-            params: { polygonPartsEntityName: 'invalid_name' as EntityIdentifier },
-            body: { filter: featureCollection<Polygon | MultiPolygon>([]) },
-          });
+        expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
+        expect(response).toSatisfyApiSpec();
 
-          expect(response.status).toBe(httpStatusCodes.BAD_REQUEST);
-          expect(response).toSatisfyApiSpec();
+        expect.assertions(2);
+      });
 
-          expect.assertions(2);
+      it('should return 400 status code if polygonPartsEntityName is an invalid value - must resolve to a valid resource identifier (no longer than 63 chars)', async () => {
+        const customConfig: IConfig = {
+          get: <T>(setting: string): T => {
+            if (setting === 'application') {
+              const defaultApplicationConfig = config.get<ApplicationConfig>(setting);
+              return {
+                ...defaultApplicationConfig,
+                entities: {
+                  ...defaultApplicationConfig.entities,
+                  parts: {
+                    namePrefix: 'very_long_prefix_',
+                    nameSuffix: '_very_long_suffix',
+                  },
+                },
+              } as unknown as T;
+            }
+            return config.get(setting);
+          },
+          // eslint-disable-next-line @typescript-eslint/unbound-method
+          has: config.has,
+          util: config.util,
+        };
+        const connectionManager = container.resolve<ConnectionManager>(ConnectionManager);
+        await connectionManager.destroy();
+        container.clearInstances();
+        const app = await getApp({
```

Comment:
is there any way to avoid re-register of those dependencies in a specific test ?

Reply:
we agreed to check if it is possible to use spy instead of mocking